### PR TITLE
 Fix false negative of `checked_conversion` lint 

### LINKED ISSUE #5675: False negative `checked_conversions` for Int::MAX
Example (playground: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f64cf36f94d0d7480c40454b89ffc06e)
```rust
#![warn(clippy::checked_conversions)]

pub fn foo(value: i64) -> Option<u32> {
    let _ = value <= (u32::max_value() as i64) && value >= 0;
    //^WARN: Checked cast can be simplified.
    let _ = value <= (u32::MAX as i64) && value >= 0; // This should also be warned but it doesn't
    Some(42)
}
```

### PR DESCRIPTION
**Pull request by lzutao**
Closes  #5675
changelog: none

### CODE REVIEW

File: tests/ui/checked_conversions.fixed [hunk 1]
```diff
@@ -1,106 +1,76 @@
...
 
-fn i64_to_u32(value: i64) -> Option<u32> {
-    if u32::try_from(value).is_ok() {
-        Some(value as u32)
-    } else {
-        None
-    }
+pub fn i64_to_u32(value: i64) {
+    let _ = u32::try_from(value).is_ok();
+    let _ = u32::try_from(value).is_ok();
```
Comment (matthiaskrgr) [inline tests/ui/checked_conversions.fixed:18]: I don't understand why this is duplicated (same for some of the other lines below).

Comment (lzutao) [reply tests/ui/checked_conversions.fixed:18]: This file is output of rustfix.

**Comment by matthiaskrgr** [comment]
Thanks!
@bors r+

**Comment by matthiaskrgr** [comment]
Ah crap, for next time, `changelog` line should be something like
"Changelog: Fix false negative of `checked_conversion` lint " :sweat_smile:

**Pull request merged**